tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -196,11 +212,30 @@ def histograms_impl(self, tag, run, downsample_to=None):
         raise ValueError('No histogram tag %r for run %r' % (tag, run))
       if downsample_to is not None and len(tensor_events) > downsample_to:
         rand_indices = random.Random(0).sample(
-            six.moves.xrange(len(tensor_events)), downsample_to)
+          six.moves.xrange(len(tensor_events)), downsample_to)
         indices = sorted(rand_indices)
         tensor_events = [tensor_events[i] for i in indices]
       events = [[e.wall_time, e.step, tensor_util.make_ndarray(e.tensor_proto).tolist()]
                 for e in tensor_events]
+    if output_format == OutputFormat.CSV:
+      csv_events = []
+      # Convert the events in a way that we can sensibly export
```

Comment:
Can you factor out some of this conversion logic into a helper function?  This overall function is already quite long.  Could be a converter for a single event -> row, or one that takes the whole list of events and returns the stringified CSV data.